syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/pubsub/SyndeoPubsubWriteSchemaTransformProvider.java

```diff
@@ -73,20 +80,49 @@ public SchemaTransform from(SyndeoPubsubWriteConfiguration configuration) {
         return new PTransform<PCollectionRowTuple, PCollectionRowTuple>() {
           @Override
           public PCollectionRowTuple expand(PCollectionRowTuple input) {
+            final Schema ERROR_SCHEMA =
+                Schema.builder()
+                    .addStringField("error")
+                    .addNullableRowField("row", input.get("input").getSchema())
+                    .build();
             SerializableFunction<Row, byte[]> fn =
                 configuration.getFormat().equals("AVRO")
                     ? AvroUtils.getRowToAvroBytesFunction(input.get("input").getSchema())
                     : JsonUtils.getRowToJsonBytesFunction(input.get("input").getSchema());
-            input
-                .get("input")
-                .apply(
-                    MapElements.into(TypeDescriptor.of(PubsubMessage.class))
-                        .via(
-                            row -> {
-                              return new PubsubMessage(fn.apply(row), Map.of());
-                            }))
+
+            PCollectionTuple outputTuple =
+                input
+                    .get("input")
+                    .apply(
+                        ParDo.of(
+                                new DoFn<Row, PubsubMessage>() {
+                                  @DoFn.ProcessElement
+                                  public void ProcessElement(
+                                      @DoFn.Element Row row, MultiOutputReceiver receiver) {
+                                    try {
+                                      receiver
+                                          .get(OUTPUT_TAG)
+                                          .output(new PubsubMessage(fn.apply(row), Map.of()));
+                                    } catch (Exception e) {
+                                      System.out.println(
+                                          "Error while parsing the element" + e.toString());
+                                      receiver
+                                          .get(ERROR_TAG)
+                                          .output(
+                                              Row.withSchema(ERROR_SCHEMA)
+                                                  .addValues(e.toString(), row)
+                                                  .build());
+                                    }
+                                  }
+                                })
+                            .withOutputTags(OUTPUT_TAG, TupleTagList.of(ERROR_TAG)));
+
+            outputTuple
+                .get(OUTPUT_TAG)
```

Comment:
just to confirm, the next line when writing to PusubIO does not return any errors currently right?

Reply:
So, it throws exceptions when it encounters an issue, it does not wrap it in an error tag and return nicely. I guess that change would need to be made on the IO side. But yeah the case where the message size is too large will occur in the PubsubIO writer

Reply:
ok, lets revisit this once IO changes are in.

Reply:
Sounds good